src/components/results/securityanalysis/security-analysis-result-utils.tsx

```diff
@@ -305,6 +318,8 @@ export const securityAnalysisTableNmKConstraintsColumnsDefinition = (
         {
             headerName: intl.formatMessage({ id: 'ContingencyId' }),
             field: 'contingencyId',
+            valueGetter: contingencyGetterValues,
```

Comment:
You can refactor all the conditions here:
{
   ...(params.data?.contingencyId && params.data?.contingencyEquipmentsIds && {valueGetter:contingencyGetterValues})
}